src/main/java/core/basesyntax/MyLinkedList.java

```
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<T> {
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return true;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            checkingIndex(index);
```

let's throw an exception here instead of calling the method checkingIndex()